### DISCUSSION TITLE
Bulk enable/duplicate entries for a new site

### DISCUSSION THREAD

**Discussion by jannisborgers** [craftcms/cms#11357] [Q&A] [answered]
We have a structure section with 300+ entries in one site and just added another site. The section’s propagation method is set to `Let each entry choose which sites it should be saved to`.

## Duplicates in environment the new site was added in
On my computer, this created „duplicates“ (in lieau of a better word) of the old entries for the new site, so all 300+ entries are existing for both sites, in sync. I can then remove entries or add new entries but have an initial base to work with.

## Doesn’t duplicate when pushed via project-config
When pushing the project-config changes to our staging server, this isn’t the case. The entries are only available for the old site. I would have to visit every entry’s detail view, pop open the site selection, and add them to the new site, in order to get the duplicated section data that we need.

## Console commands to the rescue?
I tried using the bulk resave console commands:
```
./craft resave/entries --section pages --site site_b --set status --to "=enabled"
```

… and variations of that, but get an exception `Setting read-only property: craft\elements\Entry::status`. Is there a way to add all entries in that section to the new site without having to go to each entry?

_PS: As an aside, from the docs I cannot really see what the correct status would be. What’s the difference between `disabled` (lightswitch off in the CMS UI), and `not existing for this particular site` (no lightswitch at all)_

**Comment by brandonkelly** [comment, marked as answer]
> On my computer, this created „duplicates“ (in lieau of a better word) of the old entries for the new site, so all 300+ entries are existing for both sites, in sync. I can then remove entries or add new entries but have an initial base to work with.

If the section’s Propagation Method was indeed set to “Let each entry choose which sites it should be saved to” at the time when the new site was added, this should **not** have happened. That setting leaves it up to each individual entry to choose which sites it should be saved to, and will never automatically propagate a bunch of entries to a new site.

> _PS: As an aside, from the docs I cannot really see what the correct status would be. What’s the difference between `disabled` (lightswitch off in the CMS UI), and `not existing for this particular site` (no lightswitch at all)_

If there’s a lightswitch, the entry exists in that site (regardless of whether it’s enabled or disabled); if there’s no lightswitch, the entry doesn’t exist at all for that site. It would not be possible to query the entry in that site even with `status(null)`, etc. The entry itself may as well not even exist to begin with, as far as that site is concerned.

If that’s not the behavior you’re after, maybe the simplest option here is to change your Propagation Method.

If you want to keep the current method, you can add all of the existing entries to the new site using a PHP script:

```sh
<?php

use craft\elements\Entry;

$newSiteId = 10;

$entries = Entry::find()
    ->section('pages')
    ->site('site_a')
    ->anyStatus()
    ->all();

foreach ($entries as $entry) {
    /** @var Entry $entry */
    $enabledForSite = [$entry->siteId => $entry->enabledForSite];
    foreach ($entry->getLocalized()->all() as $localized) {
        /** @var Entry $localized */
        $enabledForSite[$localized->siteId => $localized->enabledForSite];
    }
    if (!isset($enabledForSite[$newSiteId])) {
        $enabledForSite[$newSiteId] = true;
        $entry->setEnabledForSite($enabledForSite);
        Craft::$app->elements->saveElement($entry);
    }
}
```

(Replace `site_a` with the handle of a site where the entries definitely already exist in, and the `$newSiteId` value with the ID of the newly-added site.)

**Reply by jannisborgers** [reply]
In retrospect, I probably created the new site first, _then_ changed the propagation method.

> If that’s not the behavior you’re after, maybe the simplest option here is to change your Propagation Method.
> 
> If you want to keep the current method, you can add all of the existing entries to the new site using a PHP script:

I was able to achieve what we need by setting the section’s propagation method to `Save entries to all sites enabled for this section` and pushing to production, which propagated the entries to the new site, and then changing the propagation method back to `Let each entry choose which sites it should be saved to` and pushing again.

Now we have a copy of the section for our new site, which editors can change, including adding or deleting entries as they wish for the new site.

Thank you Brandon! The PHP script might help others who’re looking for something similar.
